### PR TITLE
Mixin: Update sidecar alert

### LINKED ISSUE #544: ThanosSidecarPrometheusDown error no metrics thanos_objstore_bucket_operation_failures_total
ThanosSidecar do not provide thanos_objstore_bucket_operation_failures_total by 

https://github.com/improbable-eng/thanos/blob/master/examples/alerts/alerts.md
- alert: ThanosSidecarBucketOperationsFailed
is not working

steps to reproduce:
curl http://ThanosSidecarIP:10902/metric

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <764524258@qq.com>

Fix #544

- This metrics has removed by #599, so remove this alert.
- Add new alert

### CODE REVIEW

**Comment by kakkoyun** [comment]
@daixiang0 Would you maybe consider implementing sidecar alerts and dashboards for sidecar in thanos-mixin? That one is missing.

**Comment by daixiang0** [comment]
I would like to do this :)

**Comment by kakkoyun** [comment]
Super, cool @daixiang0. Ping me if you need anything on slack :)

**Comment by daixiang0** [comment]
@kakkoyun i would file another pr for that.

**Comment by kakkoyun** [comment]
If you gonna create that PR, you will probably automate the generation of the alert examples so maybe we should close this one and focus on the upcoming PR. @daixiang0 WDYT?

**Comment by daixiang0** [comment]
It would need much time to finish while this useless alert may make users confused i think.

**Comment by daixiang0** [comment]
@kakkoyun please review.

File: mixin/thanos/alerts/sidecar.libsonnet [hunk 1]
```diff
@@ -9,6 +9,18 @@
       {
         name: 'thanos-sidecar.rules',
         rules: [
+          {
+            alert: 'ThanosSidecarPrometheusDown',
+            annotations: {
+              message: 'The prometheus which Thanos Sidecar {{$labels.job}} {{$labels.pod}} listened is down.',
```
Comment (povilasv) [inline mixin/thanos/alerts/sidecar.libsonnet]:
```suggestion
              message: 'Thanos Sidecar {{$labels.job}} {{$labels.pod}} cannot connect to Prometheus ',
```

File: mixin/thanos/alerts/sidecar.libsonnet [hunk 2]
```diff
@@ -9,6 +9,18 @@
       {
         name: 'thanos-sidecar.rules',
         rules: [
+          {
+            alert: 'ThanosSidecarPrometheusDown',
```
Comment (povilasv) [inline mixin/thanos/alerts/sidecar.libsonnet]: consider adding `for:5m` to make this less flaky

File: examples/alerts/alerts.md [hunk 3]
```diff
@@ -249,16 +249,6 @@ rules:
     impact: Prometheus configuration is not being refreshed
     action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
     dashboard: SIDECAR_URL
-- alert: ThanosSidecarBucketOperationsFailed
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply examples/alerts/alerts.md]: This metric is removed indeed.

File: mixin/thanos/alerts/sidecar.libsonnet [hunk 4]
```diff
@@ -9,6 +9,19 @@
...
+            annotations: {
+              message: 'Thanos Sidecar {{$labels.job}} {{$labels.pod}} cannot connect to Prometheus.',
```
Comment (kakkoyun) [inline mixin/thanos/alerts/sidecar.libsonnet]: You're referring to a `job` and `pod` label in alert name, these have to be exposed by your alert expression down below at line 18.

Something like
`sum by (job, pod) (thanos_sidecar_prometheus_up{%(selector)s} == 0)`

Comment (daixiang0) [reply mixin/thanos/alerts/sidecar.libsonnet]: By default, when you search a metric, it would return all label. Is it necessary?

Comment (kakkoyun) [reply mixin/thanos/alerts/sidecar.libsonnet]: Nevermind. you're right. It's not necessary in this your case. You already have a pod label.

**Comment by daixiang0** [comment]
@kakkoyun ping

File: examples/alerts/alerts.md [hunk 5]
```diff
@@ -236,39 +236,26 @@ rules:
...
   labels:
-    team: TEAM
+    severity: critical
+- alert: ThanosSidecarUnhealthy
   annotations:
-    summary: Thanos Sidecar bucket operations are failing
-    impact: We will lose metrics data if not fixed in 24h
-    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
-    dashboard: SIDECAR_URL
-- alert: ThanosSidecarGrpcErrorRate
```
Comment (kakkoyun) [inline examples/alerts/alerts.md]: @daixiang0 Let's not lose this alert, we have similar more up-to-date ones in other components' alerts.

Comment (kakkoyun) [reply examples/alerts/alerts.md]: Then we're good to merge.

**Comment by bwplotka** [comment]
Who is generating this? will `make docs` generate it? :thinking:

**Comment by kakkoyun** [comment]
@bwplotka it's `make examples` since it's where it gets updated.

https://github.com/thanos-io/thanos/blob/56a1fb6c084f11638f6f7b6532adc07dd05824b6/Makefile#L335-L337

File: examples/alerts/alerts.md (hunk 5 above)
Comment (daixiang0) [reply examples/alerts/alerts.md]: The impact and action part are only seen in this alert, is it necessary to add for all component alert?

Comment (kakkoyun) [reply examples/alerts/alerts.md]: No, I'd say it's not necessary. At this stage, these are just examples to start from.

**Pull request merged**